### DISCUSSION TITLE
Testing with the new Elgato 4k x capture card

### DISCUSSION THREAD

**Discussion by RelinquishedAll** [awawa-dev/HyperHDR#816] [Q&A]
I'm trying to achieve a setup that supports 4k60 HDR FRL VRR signal.

Using a 3080, HDFury 8K VRROOM, RPI4, ESP8266Hyperserial, SK6812, and as of today the Elgato 4K X. 

I have tried many capture cards and splitters, but VRR remains a problem. In theory this setup should work, however I am not getting any signal. 

Any ideas on what I could try?

**Comment by awawa-dev** [comment]
I doubt frl will work on Elgate because it's HDMI2.0 device (so probably it needs tmds mode).
![178155302-f062cc9a-8064-4f05-9571-1fdcb98f0b0e](https://github.com/awawa-dev/HyperHDR/assets/69086569/6a8c32cd-fb42-426c-b787-db7e15d7b787)
https://github.com/awawa-dev/HyperHDR/discussions/297#discussioncomment-3116361

Edit:
that's different model of elgato. I havent tested 4K X yet as it's newer model.

**Comment by RelinquishedAll** [comment]
The elgato 4k x is HDMI 2.1

Edit:

Sorry didn't see your edit. 

Doesn't seem to work out of the box (would've been nice but didn't expect it to)

Thank you for all your work on this project, I've been enjoying it a lot.

**Comment by awawa-dev** [comment]
Maybe you can ask Elgato support if the card supports FRL, as it would be interesting for other users too? I don't think FRL is mandatory as an HDMI2.1 devices can also revert to TMDS (but this won't happen in your case as there is an established FRL connection between source and TV and HDFury is unable to convert FRL to TMDS for one output where the grabber is connected).

**Comment by RelinquishedAll** [comment]
I'll send them an email and report back

**Comment by RelinquishedAll** [comment]
Didn't end up sending an email. Happy to report that it is working now.

![image](https://github.com/awawa-dev/HyperHDR/assets/7945709/367cd663-ecb0-4c46-8234-b7882e44684f)

info from VRROOM

Had to connect the Elgato 4k x to a windows pc with Elgato Capute Utility installed and update the firmware,  and set the resolution to 3840x2160.

Then reconnected Elgato to RPI with HyperHDR, and set the capture resolution to 3840x2160, framerate and video mode to automatic, and enabled quarter of a frame mode. Capture in visualiser is smooth and looks great. Still need to calibrate the colors a bit as the captured image does look a little pale. 

Next tests will be with various movies. I run Kodi+MPC-BC+MadVR, where MadVR switches the display modes of my LG CX to match the content. I've had many issues with this with previous splitter/capture device setups (the Elgato HD60x for example doesn't support some common movie resolution/framerate combinations. The MS2109 you blogged about works fantastic for all movie video formats, but has issues with videogames due to multiple things. Some games do weird things with the resolution through Borderless Windowed and Fullscreen settings. That, and the lack of VRR support by the MS2109).

the AVSTAR 8K@60Hz 4K@120Hz HDMI 2.1 Splitter (also released by EZCOO, same chips inside) faced similar problems, where 4k VRR HDR would work (even at 120 frames) but then would have difficulties with common movie resolution/framerate combinations. 

Elgato has the following capture information listed for the 4k x, which does seem to lack movie standards.

![image](https://github.com/awawa-dev/HyperHDR/assets/7945709/317da7dc-eb59-4f1d-928d-b4214ef9b371)

In a preliminary test I found that it does capture 4k23.976 HDR. It will take me a while to go through different media and see if the Elgato can capture it. 

Sadly now audio is not working anymore. You plug one hole and another opens..

**Reply by awawa-dev** [reply]
Thank you for the update. Sometimes USB grabbers lack popular EDID framerate settings like 23.976Hz but if they are forced to handle such signal (by HDMI splitter which cannot change the framerate for one output) it works anyway. You mean that the 3080 to TV speakers audio doesnt work? Maybe try to disable EARC as it is rather not necessary in such setup without an amplifier or I miss something.

**Reply by RelinquishedAll** [reply]
I am using a 5.1 denon receiver with this setup as well.

I'll be testing things this weekend and post an update.

**Reply by dbell669** [reply]
Is the audio issue only happening on PC or every device? If its just the PC Nvidia's 3080 may be the cause; it has issues processing 120 hz and something like Atmos at the same time. I ran into a similar issue when I was first setting up my PC and systems to the Yahama receiver, and despite several attempts they made to fix it with driver updates it never seemed to work. I ended up using my second monitor at a lower resolution, and then sending audio through that to the receiver. There's probably a more elegant solution  but it works.
